src/Dfe.Spi.EntitySquasher/Dfe.Spi.EntitySquasher.FunctionApp/Functions/GetSquashedEntity.cs

```diff
@@ -63,30 +60,102 @@ public class GetSquashedEntity
                 throw new ArgumentNullException(nameof(httpRequest));
             }
 
-            string getSquashedEntityRequestStr = null;
-            using (StreamReader streamReader = new StreamReader(httpRequest.Body))
+            // Set the context of the logger.
+            this.loggerWrapper.SetContext(httpRequest.Headers);
+
+            GetSquashedEntityRequest getSquashedEntityRequest = null;
+            try
             {
-                getSquashedEntityRequestStr = streamReader.ReadToEnd();
+                getSquashedEntityRequest = this.ParseRequest(httpRequest);
+            }
+            catch (JsonReaderException)
+            {
```

Comment:
When this goes wrong, it will not be easy to figure out why as it is never logged nor returned

Reply:
It's logged lower down - if the request is null (i.e. it couldn't be parsed because it's either null or garbage), it'll be logged as a warning (above). Any other exceptions will still be thrown.

Reply:
Sure, I see you log an error; but it lacks the detail of why it is an issue. Was it not JSON? Was one of the values not parse-able? Something else? You could obviously, with experience, work it out by looking at the payload; but why make it harder rather than just log the reason?

Reply:
@simonbu11 Fair enough, you do have a point - I've personally never found the JSON.NET parsing exceptions _useful_, but it couldn't harm to include this detail. I'll put this in!

Reply:
@simonbu11 This guy has now been tweaked.